From 2d083b47d2776a01736e4ea618f6e60db7b8b709 Mon Sep 17 00:00:00 2001
From: T S Rameshkumar <rameshkumar.t@phytecembedded.in>
Date: Fri, 16 Oct 2026 15:26:08 +0530
Subject: [PATCH 11/13] imx8m: phyboard_polis: Prune CONFIG_FEC_UIO and enable
 TI DP83867 PHY module

Remove CONFIG_FEC_UIO and non-i.MX8MM driver fragments from
android_addition_defconfig, which caused fatal modpost undefined symbol
errors for stmmac. Enable CONFIG_DP83867_PHY=m and CONFIG_RTC_DRV_RV3028=m
for phyBOARD-Polis hardware support.

Signed-off-by: T S Rameshkumar <rameshkumar.t@phytecembedded.in>
---
 imx8m/phyboard_polis/SharedBoardConfig.mk     |  2 ++
 .../phyboard_polis/android_addition_defconfig | 22 +++----------------
 2 files changed, 5 insertions(+), 19 deletions(-)

diff --git a/imx8m/phyboard_polis/SharedBoardConfig.mk b/imx8m/phyboard_polis/SharedBoardConfig.mk
index 1a102cc7..ad89aeac 100644
--- a/imx8m/phyboard_polis/SharedBoardConfig.mk
+++ b/imx8m/phyboard_polis/SharedBoardConfig.mk
@@ -63,8 +63,10 @@ BOARD_VENDOR_KERNEL_MODULES += \
     $(KERNEL_OUT)/drivers/mxc/hantro_845_h1/hx280enc.ko \
     $(KERNEL_OUT)/drivers/mxc/hantro_v4l2/vsiv4l2.ko \
     $(KERNEL_OUT)/drivers/rtc/rtc-snvs.ko \
+    $(KERNEL_OUT)/drivers/rtc/rtc-rv3028.ko \
     $(KERNEL_OUT)/drivers/net/phy/at803x.ko \
     $(KERNEL_OUT)/drivers/net/phy/realtek.ko \
+    $(KERNEL_OUT)/drivers/net/phy/dp83867.ko \
     $(KERNEL_OUT)/drivers/net/ethernet/freescale/fec.ko
 endif
 
diff --git a/imx8m/phyboard_polis/android_addition_defconfig b/imx8m/phyboard_polis/android_addition_defconfig
index 3528673d..039d0435 100644
--- a/imx8m/phyboard_polis/android_addition_defconfig
+++ b/imx8m/phyboard_polis/android_addition_defconfig
@@ -1,21 +1,5 @@
-CONFIG_DRM_MXSFB=m
-CONFIG_DWMAC_IMX8=m
-CONFIG_MXC_VIDEO_WAVE6=m
-CONFIG_VIDEO_AMPHION_VPU=m
-CONFIG_DRM_IMX_DCNANO=m
-# CONFIG_HAVE_IMX_RPMSG is not set
-CONFIG_DRM_IMX95_DPU=m
-CONFIG_DRM_IMX_DCSS=m
-CONFIG_OPTEE=m
-CONFIG_FEC_UIO=m
-CONFIG_IMX_LCDIFV3_CORE=m
-CONFIG_IMX_DPU_CORE=m
-CONFIG_DRM_NWL_MIPI_DSI=m
-CONFIG_IMX8_DPRC=m
-CONFIG_IMX_IRQSTEER=m
-CONFIG_DRM_CDNS_MHDP=m
-CONFIG_DRM_IMX_CDNS_MHDP=m
-CONFIG_FB_MXC_EINK_V2_PANEL=m
-CONFIG_DRM_IMX_RPMSG=m
+# PHYTEC phyBOARD-Polis i.MX8MM additions
+CONFIG_DP83867_PHY=m
+CONFIG_RTC_DRV_RV3028=m
 CONFIG_LOCALVERSION="-g758ad1ac4e8e-PHYTEC-Android15"
 # CONFIG_LOCALVERSION_AUTO is not set
-- 
2.34.1

